src/service/chat.js

```diff
@@ -206,19 +206,18 @@ const moveChatToAnon = async ({
   }
 };
 
-const initChatFromPost = async ({source, id}) => {
+const initChatFromPost = async ({source, id, postId}) => {
   let payload = {source};
   switch (source) {
     case 'post':
       payload = {...payload, postId: id};
       break;
     case 'comment':
-      payload = {...payload, commentId: id};
+      payload = {...payload, commentId: id, postId};
       break;
     default:
       break;
   }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are logical and consistent. The `postId` is now being passed in the payload for both 'post' and 'comment' cases, which seems to be a necessary addition considering the context of initiating a chat from a post or comment. However, it would be good to add error handling for the case when `source` is neither 'post' nor 'comment'. Currently, the function will proceed with an empty payload if `source` doesn't match either of these cases, which could lead to unexpected behavior.

```diff
+    default:
+      throw new Error(`Invalid source: ${source}`);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->